### ISSUE TITLE
create mongoose models

### ISSUE THREAD

**Issue by vesparny** [getPoseidon/Poseidon#12]
We need to create the basic mongoose models.
@spacemonkey could you help on this?

**Comment by spacemonkey** [comment]
YES! And I'm happy with using mongoose, but think we need to be minimal on our dependency. You can get lost in infinite population loops for instance, and you end up spending more time getting around the ODM than the time it was supposed to be saving you :-)

That said, we should take discussion of base models probably to the wiki, so we can talk about what is actually going to be in 1.0...

**Comment by vesparny** [comment]
we can create an abstraction layer, exposing just an api,

getPost, createPost ...etc...

in this way we can encapsulate and isolate the mongoose logic behind our
layer

**Comment by spacemonkey** [comment]
Here's something I cannot solve gracefully in the Node world - say there's an addComment in both Post and Page controllers with almost the same logic... How do you want to keep that code DRY?

**Comment by vesparny** [comment]
You can simply create a service call commentService, and inject it into
your controllers using our IoC mechanism
